source/common/config/watch_map.h

```diff
@@ -109,7 +105,15 @@ class WatchMap : public UntypedConfigUpdateCallbacks, public Logger::Loggable<Lo
                                      Watch* watch);
 
   // Returns the union of watch_interest_[resource_name] and wildcard_watches_.
-  absl::flat_hash_set<Watch*> watchesInterestedIn(const std::string& resource_name);
+  absl::flat_hash_set<Watch*> watchesInterestedIn(const std::string& resource_name,
+                                                  const bool use_namespace_matching);
+
+  // Returns the namespace part (if there's any) in the resource name.
+  std::string namespaceFromName(const std::string& resource_name) {
```

Comment:
Should this be inside the `.cc` as an anonymouns namespace helper?

Reply:
done